### DISCUSSION TITLE
Need export option

### DISCUSSION THREAD

**Discussion by vishwa-ims** [RevealBi/Reveal.Sdk#62] [ASP.NET Core]
Hi,

We are using ASP.NET Core SDK and we need to give button in our UI to export the dashboard. Currently we can export the dashboard by clicking on export button that we do have on dashboard screen of SDK but we need to give this custom button outside dashboard

**Comment by brianlagunas** [comment]
At this time we do not have a public API do perform an export programatically. However, as a workaround, you could use a javascript work-around to achieve this.

You can call these methods off of the reveal view instance to perform your exports

```js
    exportToExcel(): void {
        this._revealView._dashboardView.exportToExcel();
    }

    exportToImage(showDialog: boolean = true): void | Promise<Element | null> {

        if (showDialog) {
            this._revealView._dashboardView.exportImage();
            return;
        }

        return this._revealView.toImage();
    }

    exportToPdf(): void {
        this._revealView._dashboardView.exportToFormat("pdf");
    }

    exportToPowerPoint(): void {
        this._revealView._dashboardView.exportToFormat("pptx");
    }
```

**Reply by vishwa-ims** [reply]
@brianlagunas Is this the code that we can use for server side export ?
https://help.revealbi.io/web/server-export?code=aspnet

**Reply by brianlagunas** [reply]
Yes, that is server-side export. Keep in mind that will not have any of the end-user applied filters or statistical functions. Those are all applied client-side.

**Reply by vishwa-ims** [reply]
Thanks Brian. Will this export whole dashboard or do we have any option to export any specific visualisation ?

**Reply by brianlagunas** [reply]
The entire dashboard. There is currently no option to export a single visualization.
